### PR TITLE
camera-server: Add ability to support taking image in video mode and video in image mode

### PR DESCRIPTION
**Pull request by jnomikos**
PR associated with this change:
https://github.com/mavlink/MAVSDK/pull/2351

### CODE REVIEW

**Review by JonasVautherin — COMMENTED** [review]
Couldn't those go in the `SetInformation` function above? Since it affects the CAMERA_INFORMATION message :thinking:

**Comment by jnomikos** [comment]
> Couldn't those go in the `SetInformation` function above? Since it affects the CAMERA_INFORMATION message 🤔

Yeah I could put it in `SetInformation`. The reason I didn't was because it is probably rare for a user to want to set this. But I agree it makes more sense to be there.

File: protos/camera_server/camera_server.proto [hunk 1]
```diff
@@ -515,6 +521,20 @@ message RespondTrackingOffCommandResponse {
...
 
+message SupportImageInVideoModeRequest {
+    bool support = 1; // True if the camera supports capturing images while in video mode
+}
+message SupportImageInVideoModeResponse {
+    CameraServerResult camera_server_result = 1;
+}
+
+message SupportVideoInImageModeRequest {
+    bool support = 1; // True if the camera supports capturing video while in image mode
```
Comment (julianoes) [inline protos/camera_server/camera_server.proto:532]:
```suggestion
    bool is_supported = 1; // True if the camera supports capturing video while in image mode
```

File: protos/camera_server/camera_server.proto [hunk 2]
```diff
@@ -515,6 +521,20 @@ message RespondTrackingOffCommandResponse {
     CameraServerResult camera_server_result = 1; // The result of sending the response.
 }
 
+message SupportImageInVideoModeRequest {
+    bool support = 1; // True if the camera supports capturing images while in video mode
```
Comment (julianoes) [inline protos/camera_server/camera_server.proto:525]:
```suggestion
    bool is_supported = 1; // True if the camera supports capturing images while in video mode
```

**Review by julianoes — COMMENTED** [review]
I think that makes sense. It would be nice if this came implicitly out of the camera subscriptions but I don't think there is an easy way to do so.

And thanks for the PR!

**Review by julianoes — APPROVED** [review]
I think that's easier and makes more sense.

**Pull request merged**